### ISSUE TITLE
[Datepicker] - Can not delete selected date value by press button "delete" on keyboard

### ISSUE THREAD

**Issue by phanthinh** [rsuite/rsuite#2762]
### What version of rsuite are you using?

5.12.0

### What version of React are you using?

17.0.2

### What browser are you using?

Chrome, Safari

### Describe the Bug

When I selected a date in date picker and remove it by keyboard, the date show again when I focus out date picker.

![1](https://user-images.githubusercontent.com/5910795/192147687-b059692f-ca32-4334-a50e-e49f0cc1fd3f.gif)

### Expected Behavior

Step 1: Choose a date
Step 2: Click on the input to selected date
Step 3: click "delete" button on keyboard, the date removed
Step 4: CLick out date picker, the date removed.

**Comment by simonguo** [comment]
You can use `onBlur` to check for changes in value to achieve your application requirements.

```jsx
const App = () => {
  const [value, setValue] = React.useState(null);
  return (
    <DatePicker
      value={value}
      onChange={setValue}
      onBlur={(event) => {
        if (event.target.value === "") {
          setValue(null);
        }
      }}
    />
  );
};
```

**Comment by simonguo** [comment]
Fixed in v5.72.0
